File: benchmarks/fixtures/polyglot-services/src/UserService.hs
module Main where

data User = User { name :: String, email :: String }

display :: User -> String
display (User n e) = n ++ " <" ++ e ++ ">"

main :: IO ()
main = do
    let user = User "Alice" "alice@example.com"
    putStrLn (display user)

File: benchmarks/fixtures/polyglot-services/src/user_service.clj
(ns user-service.core
    (:require [clojure.string :as str]))

(defn display [name email]
    (str name " <" email ">"))

(defn create [params]
    {:name (:name params) :email (:email params)})

(def user (create {:name "Charlie" :email "charlie@example.com"}))
(println (display (:name user) (:email user)))